### PR TITLE
Update template to use TFM-based sources; enabled latest lang features

### LINKED ISSUE #5074: Update C# templates with global usings
We can simplify our templates with [global usings](https://youtu.be/GJ_PaRNDe9E?t=1079) and have them implemented similar to this:

```diff
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework Condition="'$(TargetFrameworkOverride)' == ''">FrameworkParameter-windows</TargetFramework>
    <TargetFramework Condition="'$(TargetFrameworkOverride)' != ''">TargetFrameworkOverride-windows</TargetFramework>
    <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.WinFormsApplication1</RootNamespace>
    <LangVersion Condition="'$(langVersion)' != ''">$(ProjectLanguageVersion)</LangVersion>
    <Nullable Condition="'$(Nullable)' == 'true'">enable</Nullable>
    <UseWindowsForms>true</UseWindowsForms>
  </PropertyGroup>

+ <ItemGroup>
+   <Compile Include=".usings" />
+ </ItemGroup>

</Project>
```

A sample .using file:
```diff
global using System;
global using System.Collections.Generic;
global using System.Drawing;
global using System.Linq;
global using System.Threading.Tasks;
global using System.Windows.Forms;
```

This list of usings looks sensible, but it is up for a discussion.

Credits for this idea go to https://github.com/khalidabuhakmeh/CsharpTenFeatures.

At some point in the future when https://github.com/dotnet/aspnetcore/issues/32451 settles, we could follow in their steps and bake all typical Windows Forms usings into the Windows Desktop SDK.

### PR DESCRIPTION
**Pull request by RussKie**
(cherry picked from commit ee658c4156fe099a5b48d4b61d11f996a468ca4f / #5547)

Fixes #5074

## Proposed changes

- Enable implicit global usigns
- Enable file-based namespaces

## Regression? 

- No, new functionality

## Risk

- Minimal

## Test methodology 

Manually create various apps and verify the expected content

```
dotnet new -i C:\Development\winforms\pkg\Microsoft.Dotnet.WinForms.ProjectTemplates\content\WinFormsApplication-CSharp
cd .. && md template1 && cd template1 && dotnet new winforms  && code . 
cd .. && md template2 && cd template2 && dotnet new winforms -f net6.0 && code . 
cd .. && md template3 && cd template3 && dotnet new winforms -f net5.0 && code . 
cd .. && md template4 && cd template4 && dotnet new winforms -f netcoreapp3.1 && code . 

dotnet new -i C:\Development\winforms\pkg\Microsoft.Dotnet.WinForms.ProjectTemplates\content\WinFormsControlLibrary-CSharp
cd .. && md template12 && cd template12 && dotnet new winformscontrollib && code . 
cd .. && md template22 && cd template22 && dotnet new winformscontrollib -f net6.0 && code . 
cd .. && md template32 && cd template32 && dotnet new winformscontrollib -f net5.0 && code . 
cd .. && md template42 && cd template42 && dotnet new winformscontrollib -f netcoreapp3.1 && code . 

dotnet new -i C:\Development\dotnet-winforms\pkg\Microsoft.Dotnet.WinForms.ProjectTemplates\content\WinFormsLibrary-CSharp
cd .. && md template01 && cd template01 && dotnet new winformslib  && code . 
cd .. && md template21 && cd template21 && dotnet new winformslib -f net6.0 && code . 
cd .. && md template31 && cd template31 && dotnet new winformslib -f net5.0 && code . 
cd .. && md template41 && cd template41 && dotnet new winformslib -f netcoreapp3.1 && code .

```
 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5609)

### CODE REVIEW

**Comment by RussKie** [comment]
@Pilchie seeking approval to take into RC2.

@DamianEdwards this essentially wraps our templating story.

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/Company.WinFormsApplication1.csproj.user [hunk 1]
```diff
@@ -1,8 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="Current" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-    <ItemGroup>
-        <Compile Update="Form1.cs">
-            <SubType>Form</SubType>
```
Comment (Pilchie) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/Company.WinFormsApplication1.csproj.user:5]: Why is this deleted?  Isn't this needed for the designer to be available immediately after project creation?

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Company.WinFormsApplication1.csproj [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>WinExe</OutputType>
+    <TargetFramework Condition="'$(TargetFrameworkOverride)' == ''">FrameworkParameter-windows</TargetFramework>
+    <TargetFramework Condition="'$(TargetFrameworkOverride)' != ''">TargetFrameworkOverride-windows</TargetFramework>
+    <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.WinFormsApplication1</RootNamespace>
+    <LangVersion Condition="'$(langVersion)' != ''">$(ProjectLanguageVersion)</LangVersion>
```
Comment (Pilchie) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Company.WinFormsApplication1.csproj:8]: I don't think we want this to show up in the project file.

Comment (Pilchie) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Company.WinFormsApplication1.csproj:8]: Also, has a lower case `l`

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Form1.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+#endif
+#if (csharpFeature_FileScopedNamespaces)
+namespace Company.WinFormsApplication1;
```
Comment (Pilchie) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Form1.cs:14]: I thought the decision was not to use these in the templates, and have the VS option apply formatting to decide which form to use?

**Review by Pilchie — COMMENTED** [review]
Need @DamianEdwards' signoff here.

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Form1.cs (hunk 3 above)
Comment (DamianEdwards) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Form1.cs:14]: For templates that aren't splitting their identity based on TFM and accept a language version parameter (like these), yes, I believe that was the decision.

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/Company.WinFormsApplication1.csproj.user (hunk 1 above)
Comment (RussKie) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/Company.WinFormsApplication1.csproj.user:5]: This looks like a bad merge, thank you

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Form1.cs (hunk 3 above)
Comment (RussKie) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Form1.cs:14]: These changes were suggested to us by the dotnet/templating team and aligns with changes made in that repo (e.g. https://github.com/dotnet/templating/pull/3454).

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Company.WinFormsApplication1.csproj (hunk 2 above)
Comment (RussKie) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Company.WinFormsApplication1.csproj:8]: This is a command line argument and aligns with this:
https://github.com/dotnet/winforms/blob/07f6de36d1dc4e86daccf4686ab222b950473673/pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/template.json#L52-L59

The command line args are case sensitive:
image

There's an open tracking issue for this https://github.com/dotnet/templating/issues/3237

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Form1.cs (hunk 3 above)
Comment (DamianEdwards) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/net6.0/Form1.cs:14]: @Pilchie @KathleenDollard was it that item templates would keep using block-scoped namespace declarations, but the project templates that had an option to change the language version would selectively apply file-scoped when the version was high-enough (or not specified) perhaps?

**Comment by Pilchie** [comment]
@davidwengier - do you know what the conclusion was for templates and file scoped namespaces?

**Comment by davidwengier** [comment]
> @davidwengier - do you know what the conclusion was for templates and file scoped namespaces?

In current dogfood builds, Roslyn will reformat namespaces (and other things) for any new file that is added, based on a users preferences.

**Comment by RussKie** [comment]
So any further objections?

**Comment by DamianEdwards** [comment]
@RussKie no I think you're good to go.
